libs/ui/src/components/MediaItem/MediaItem.vue

```diff
@@ -110,7 +106,7 @@ export default {
   },
   methods: {
     async updateComponent() {
-      if (this.animationSrc && !this.mimeType) {
+      if (this.animationSrc) {
```

Comment:
This will overfetch on `HEAD` request

Reply:
Do you have any idea for an alternative fix?

Except, we can store that mime-type stuff from the indexer side to pass that mime-type as a prop. At the moment, we can't rely on `meta.type`

Reply:
+1 to store it on indexer so we can query different mime type

Reply:
> Do you have any idea for an alternative fix?


1.
- https://github.com/kodadot/loligo/issues/44

2.
- https://github.com/kodadot/research/issues/103

> we can't rely on meta.type

Why so? 
80% of content (except rmrk2) was minted via kodadot. 
So a root isssue can be also hidden in UI

Reply:
> * [Fetch mime-type of animation_url loligo#44](https://github.com/kodadot/loligo/issues/44)

> Why so?

Need mime type for both media and animated media. in this [item](https://deploy-preview-6599--koda-canary.netlify.app/bsx/gallery/3030053998-2), both media and animated media mime type is `image/jpeg`. but, in metadata is `application/octet-stream`
![image](https://github.com/kodadot/nft-gallery/assets/734428/be24cb52-5c0c-461a-94c1-74ef285f4c67)


> So a root isssue can be also hidden in UI

then, how do we check that the item mime type is correct or wrong in the current state in our UI?

can someone invite me to this repo? https://github.com/kodadot/research/issues/103 I can't open the link

Reply:
> Question Is why don't we fetch that directly from subsquid?

afaik, mostly this is what we already do now
1. check `mime.type` from subsquid
2. if not exists, fetch mime type from `image` or `animationUrl`

for example, that item is currently `null` from subsquid

![image](https://github.com/kodadot/nft-gallery/assets/734428/7e225b91-cde8-4974-b954-d044d223111a)

I found out that this is similar case on rmrk2 https://github.com/kodadot/workers/issues/95. that's why I said can't rely on `metadata` or `meta.type`. and, if we want to address overfetch issue, put the "correct" mimetype to indexer

> Basically make D1 with metadata table (name, description, attributes, image ...)

if your concern is overfetch, for this case is overkill, imo. because we can solve it by request with HEAD method. and the different with this is only endpoint change

for now, I will update it to default render image component. but, if this case possible https://github.com/kodadot/nft-gallery/issues/5498#issuecomment-1669515097 it will render incorrectly

Reply:
One more edge case https://github.com/kodadot/nft-gallery/issues/6618. the creator puts `image` and `metadata` on the s3 bucket. Usually, the other creator uses IPFS.

`metadata` stuff is user define so that the output can be anything. Based on that, I assume that I am more convinced that we can't rely on the `type` attribute from metadata and even the `image` attribute.

Reply:
updated, render image by default

notes: nft creator can put other than a static image in their `image` or `mediaUri` attributes on metadata